src/baseTrie.ts

```diff
@@ -26,11 +26,11 @@ interface Path {
   stack: TrieNode[]
 }
 
-type FoundNodeFunction = (
+export type FoundNodeFunction = (
   nodeRef: Buffer,
   node: TrieNode,
   key: Nibbles,
-  walkController: any
+  walkController: WalkStrategy
```

Comment:
would it make sense to name the class WalkController? (i am curious why you chose WalkStrategy)

Reply:
Yeah you are right, the original `WalkController` is a bit more appropriate. The consumer of it would actually provide the "strategy" to the "controller".

Reply:
Renamed this in 3a7fe21